packages/core/src/modules/connections/handlers/DidExchangeRequestHandler.ts

```diff
@@ -39,16 +44,39 @@ export class DidExchangeRequestHandler implements MessageHandler {
       throw new AriesFrameworkError('Unable to process connection request without senderKey or recipientKey')
     }
 
-    if (!message.thread?.parentThreadId) {
+    const parentThreadId = message.thread?.parentThreadId
+
+    if (!parentThreadId) {
       throw new AriesFrameworkError(`Message does not contain 'pthid' attribute`)
     }
-    const outOfBandRecord = await this.outOfBandService.findByCreatedInvitationId(
-      messageContext.agentContext,
-      message.thread.parentThreadId
-    )
 
+    const createOobRecord = async (did: string) => {
+      const outOfBandInvitation = new OutOfBandInvitation({
+        id: message.threadId,
+        label: '',
+        services: [did],
+        handshakeProtocols: [HandshakeProtocol.DidExchange],
+      })
+
+      const outOfBandRecord = new OutOfBandRecord({
+        role: OutOfBandRole.Sender,
+        state: OutOfBandState.AwaitResponse,
+        reusable: true,
+        autoAcceptConnection: this.connectionsModuleConfig.autoAcceptConnections,
+        isImplicitInvitation: true,
+        outOfBandInvitation,
+      })
+
+      await this.outOfBandService.save(messageContext.agentContext, outOfBandRecord)
+      this.outOfBandService.emitStateChangedEvent(messageContext.agentContext, outOfBandRecord, null)
+      return outOfBandRecord
+    }
```

Comment:
Creating oob records is more the domain of the oob module. Maybe we can add this to the out of band service and reuse it for both the connection and did exchange protocols? I think that will make it easier if we need to change the oob implementation

Reply:
Agree, I'll move this logic to oob and call it from the handlers.